Analysis/Tasks/PWGDQ/v0selector.cxx

```diff
@@ -42,19 +44,275 @@ using namespace o2::framework;
+  //void init(o2::framework::InitContext&)
+  //{
+  //}
+  Produces<aod::ReducedV0s> v0Gamma;
+
+  enum { // Reconstructed V0
+    kUndef = -1,
+    kGamma = 0,
+    kK0S = 1,
+    kLambda = 2,
+    kAntiLambda = 3
+  };
+
+  float alphav0(const array<float, 3>& ppos, const array<float, 3>& pneg)
+  {
+    std::array<float, 3> pv0 = {ppos[0] + pneg[0], ppos[1] + pneg[1], ppos[2] + pneg[2]};
+    float momTot = RecoDecay::P(pv0);
+    float lQlNeg = RecoDecay::dotProd(pneg, pv0) / momTot;
+    float lQlPos = RecoDecay::dotProd(ppos, pv0) / momTot;
+    return (lQlPos - lQlNeg) / (lQlPos + lQlNeg); //longitudinal momentum asymmetry
+  }
+
+  float qtarmv0(const array<float, 3>& ppos, const array<float, 3>& pneg)
+  {
+    std::array<float, 3> pv0 = {ppos[0] + pneg[0], ppos[1] + pneg[1], ppos[2] + pneg[2]};
+    float momTot2 = RecoDecay::P2(pv0);
+    float dp = RecoDecay::dotProd(pneg, pv0);
+    return std::sqrt(RecoDecay::P2(pneg) - dp * dp / momTot2); //qtarm
+  }
+
+  float phivv0(const array<float, 3>& ppos, const array<float, 3>& pneg, const int cpos, const int cneg, const float bz)
+  {
+    // implementation taken from AliDielectronPair.cxx
+    float px1 = -9999., py1 = -9999., pz1 = -9999.;
+    float px2 = -9999., py2 = -9999., pz2 = -9999.;
+
+    if (cpos * cneg > 0.) { // Like Sign
+      if (bz < 0) {         // inverted behaviour
+        if (cpos > 0) {
+          px1 = ppos[0];
+          py1 = ppos[1];
+          pz1 = ppos[2];
+          px2 = pneg[0];
+          py2 = pneg[1];
+          pz2 = pneg[2];
+        } else {
+          px1 = pneg[0];
+          py1 = pneg[1];
+          pz1 = pneg[2];
+          px2 = ppos[0];
+          py2 = ppos[1];
+          pz2 = ppos[2];
+        }
+      } else {
+        if (cpos > 0) {
+          px1 = pneg[0];
+          py1 = pneg[1];
+          pz1 = pneg[2];
+          px2 = ppos[0];
+          py2 = ppos[1];
+          pz2 = ppos[2];
+        } else {
+          px1 = ppos[0];
+          py1 = ppos[1];
+          pz1 = ppos[2];
+          px2 = pneg[0];
+          py2 = pneg[1];
+          pz2 = pneg[2];
+        }
+      }
+    } else {        // Unlike Sign
+      if (bz > 0) { // regular behaviour
+        if (cpos > 0) {
+          px1 = ppos[0];
+          py1 = ppos[1];
+          pz1 = ppos[2];
+          px2 = pneg[0];
+          py2 = pneg[1];
+          pz2 = pneg[2];
+        } else {
+          px1 = pneg[0];
+          py1 = pneg[1];
+          pz1 = pneg[2];
+          px2 = ppos[0];
+          py2 = ppos[1];
+          pz2 = ppos[2];
+        }
+      } else {
+        if (cpos > 0) {
+          px1 = pneg[0];
+          py1 = pneg[1];
+          pz1 = pneg[2];
+          px2 = ppos[0];
+          py2 = ppos[1];
+          pz2 = ppos[2];
+        } else {
+          px1 = ppos[0];
+          py1 = ppos[1];
+          pz1 = ppos[2];
+          px2 = pneg[0];
+          py2 = pneg[1];
+          pz2 = pneg[2];
+        }
+      }
+    }
```

Comment:
Is this entire structure needed? The only place where the order of the two vectors matter is in the vector product and there you have just a binary choice (+ or - since the vector product is not commutative so you obtain two opposite vectors, depending on the order in which you apply the product). So i think that the big structure above is not really needed. The sign can be obtained from just the combination of bz, cpos and cpos*cneg.